gradle-jdks-certs/src/main/resources/gradle-jdk-resolver.sh

```diff
@@ -0,0 +1,128 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+##############################################################################
+#
+#   Gradle jdk set up script for POSIX generated by gradle-jdks.
+#
+#   This script does the following:
+#   (1) Downloads the corresponding JDK distribution based on the majorVersion=`gradle/gradle-jdk-major-version.txt`
+#     and the distribution_url=`gradle/jdks/${majorVersion}/${os}/${arch}/download_url.txt`
+#   (2) Installs the distribution in `$HOME/.gradle/gradle-jdks/${local_path}` based on the
+#     local_path=`gradle/jdks/${majorVersion}/${os}/${arch}/local_path.txt`
+#   (3) Calls the java class `JdkSpecCertSetup` that will set up the certificates for the locally installed distribution
+#   (4) Sets up the JAVA_HOME env variable to the currently installed JDK
+#
+#
+#   Important for running:
+#   This script requires all of these POSIX shell features:
+#         * functions;
+#         * expansions «$var», «${var}», «${var%suffix}», and «$( cmd )»;
+#         * various built-in commands including «command» and «set».
+#
+##############################################################################
+
+tmp_work_dir=$(mktemp -d)
+
+die () {
+    echo
+    echo "$*"
+    echo
+    # TODO(crogoz): Cleaning up
+    echo rm -rf "$tmp_work_dir"
+    exit 1
+} >&2
+
+# OS specific support
+case "$( uname )" in                          #(
+  Linux* )          os_name="linux"  ;;       #(
+  Darwin* )         os_name="macos"  ;;       #(
+  * )               die "ERROR Unsupported OS: $( uname )" ;;
+esac
+
+if [ "$os_name" = "linux" ]; then
+    if [ -f /etc/alpine-release ]; then
+      os_name="alpine-linux"
+    fi
+fi
+
+# Arch specific support, see: gradle-jdks:com.palantir.gradle.jdks.CurrentArch.java
+case "$(uname -m)" in                         #(
+  x86-64* )       arch_name="x86-64"  ;;      #(
+  x64* )          arch_name="x86-64"  ;;      #(
+  amd64* )        arch_name="x86-64"  ;;      #(
+  arm64* )        arch_name="aarch64"  ;;     #(
+  arm* )          arch_name="aarch64"  ;;     #(
+  aarch64* )      arch_name="aarch64"  ;;     #(
+  x86* )          arch_name="x86"  ;;         #(
+  i686* )         arch_name="x86"  ;;         #(
+  * )             die "ERROR Unsupported architecture: $( uname -m )" ;;
+esac
+
+read -r major_version < gradle/gradle-jdk-major-version.txt
+read -r distribution_url < gradle/jdks/"$major_version"/$os_name/$arch_name/download-url.txt
+read -r distribution_local_path < gradle/jdks/"$major_version"/$os_name/$arch_name/local-path.txt
```

Comment:
nit: ending things in `.txt` makes me think of windows, it doesn't really matter that much but I'd just leave off the `.txt` entirely.